src/nspawn/nspawn-network.c

```diff
@@ -183,8 +207,10 @@ int setup_veth(const char *machine_name,
 
         /* Use two different interface name prefixes depending whether
          * we are in bridge mode or not. */
-        snprintf(iface_name, IFNAMSIZ - 1, "%s-%s",
-                 bridge ? "vb" : "ve", machine_name);
+        n = strappend(bridge ? "vb-" : "ve-", machine_name);
+
+        shorten_ifname(n);
+        snprintf(iface_name, IFNAMSIZ, "%s", n);
```

Comment:
strncpy() looks like the better approach here...

Reply:
Should be safe although it wouldn't add a null terminator. The source should always be shorter than `IFNAMSIZ`. Fixed with next push.